misc/LADYBUG_USER_STORIES.md

```diff
@@ -3,51 +3,152 @@ Ladybug User Stories
-I can rerun a report. This can be with stubbed responses of external systems or without stubbing | ladybug-ff-cypress-test
-I can rerun both in the test tab and in the debug tab | ladybug-ff-cypress-test
-When a rerun produces a different result, I can compare the original and the new result.
-I can configure irrelevant differences (e.g. time stamps) to be ignored | ladybug-ff-cypress-test
-In the test tab, I can select multiple reports to be rerun in one go | ladybug-ff-cypress-test
-I can download reports and upload them later, allowing them to be backed up | ladybug-ff-cypress-test
-Downloading and uploading reports should be available both in the debug tab and the test tab | ladybug-ff-cypress-test
-
-Reports in progress
--------------------
-
-Story | Test plan
------ | ---------
-Each message the F!F is currently processing results in a report-in-progress. I want to see the number of reports-in-process | ladybug-ff-cypress-test
-If a report remains in-process for more than five minutes, I want to see an alert in Ladybug | ladybug-ff-cypress-test
-
-Security and storage
---------------------
+Vision
+------
+
+Ladybug is a tool to debug and test Java applications that process messages. It is included in the Frank!Framework. This document provides user stories for Ladybug as a part of the Frank!Framework.
+
+Ladybug is not limited to work with the Frank!Framework; it is a tool that can be used with other Java applications as well. The Java application that uses Ladybug is expected to receive or retrieve messages. Each message undergoes a series of transformations to obtain an output message. The Java application has access to a class **TestTool** that is provided by the Ladybug dependency. During the transformation, the Java application uses the methods of class TestTool to produce **checkpoints**. The Java application should also call TestTool to create checkpoints for the input message and the output message. For each input message, the Java application should create a unique **correlation id** that is provided to TestTool for each created checkpoint. TestTool groups all checkpoints by correlation id into a **report**. A **report** thus describes all transformations applied to one input message.
+
+The Frank!Framework creates a Ladybug checkpoint for each pipe and each pipeline. Frank developers can thus see how each pipe transforms the message going through the pipeline.
+
+Ladybug can be used as a debug tool. Ladybug provides a user interface that shows a table of all captured reports. When the user clicks a report, it is opened in a tree view. Multiple reports can be shown simultaneously in the tree view. Each checkpoint of each report is a node in the tree. Collapsing and expanding nodes is described in more details in the user stories below. When you click a node, the corresponding message inside the checkpoint is shown.
```

Comment:
This part is removed in new ladybug UI. Only one report at a time in treeview. Comparing reports is in the other tab (test/compare?)

Reply:
Do we want only one report at the time in the tree view? Does it have added value if there can be multiple reports in the tree view simultaneously?

Reply:
Ik heb de tekst over multiple reports hier weggehaald. Dit is tekst om de context te schetsen en bevat nog niet de eigenlijke user stories. Of er meerdere rapporten tegelijk open kunnen zijn of niet is iets wat in een specifieke user story hoort. Die user story zal ik schrijven als we het allemaal eens zijn over wat we willen.